docs/tutorials/developing-with-invenio.rst

```diff
@@ -5,5 +5,339 @@
+
+The ``utils.py`` file contains all helper functions of our module,
+so let's write the first utility that will create a record.
+
+In ``invenio_foo/utils.py``
+
+.. code-block:: python
+
+    """Utils module."""
+    from __future__ import absolute_import, print_function
+
+    import uuid
+
+    from flask import current_app
+
+    from invenio_db import db
+    from invenio_indexer.api import RecordIndexer
+    from invenio_pidstore import current_pidstore
+    from invenio_records.api import Record
+
+
+    def create_record(data):
+        """Create a record.
+
+        :param dict data: The record data.
+        """
+        indexer = RecordIndexer()
+        # create uuid
+        rec_uuid = uuid.uuid4()
+        # add the schema
+        data["$schema"] = \
+            current_app.extensions['invenio-jsonschemas'].path_to_url(
+                'records/custom-record-v1.0.0.json'
+            )
+        # create PID
+        current_pidstore.minters['recid'](rec_uuid, data)
+        # create record
+        created_record = Record.create(data, id_=rec_uuid)
+        db.session.commit()
+
+        # index the record
+        indexer.index(created_record)
+
+5. Create the custom-record JSON Schema
++++++++++++++++++++++++++++++++++++++++
+
+Our records can use a custom schema. To define and use this schema,
+we create the ``custom-record-v1.0.0.json`` file inside the ``records``
+folder of your data model project (``my-datamodel`` from the Quickstart
+tutorial :ref:`build-data-model`).
+
+In ``my-datamodel/my-datamodel/jsonschemas/records/custom-record-v1.0.0.json``:
+
+.. code-block:: json
+
+    {
+      "$schema": "http://json-schema.org/draft-04/schema#",
+      "id": "http://localhost/schemas/records/custom-record-v1.0.0.json",
+      "additionalProperties": true,
+      "title": "my-datamodel v1.0.0",
+      "type": "object",
+      "properties": {
+        "title": {
+          "description": "Record title.",
+          "type": "string"
+        },
+        "description": {
+          "description": "Record description.",
+          "type": "string"
+        },
+        "id": {
+          "description": "Invenio record identifier (integer).",
+          "type": "string"
+        }
+      },
+      "required": [
+        "title",
+        "description"
+      ]
+    }
+
+
+Demo time
+---------
+
+Let's now see our Invenio module in action after it has been integrated in our
+Invenio instance.
+
+First, install the new invenio-foo module in the virtual enviroment of the
+Invenio instance:
+
+.. code-block:: console
+
+    $ pipenv shell  # activate the app virtual env
+    (my-site) $ cd ../invenio-foo
+    (my-site) $ pip install --editable .
+
+Then, if you've followed the steps in the :ref:`quickstart` guide, you can go to the
+instance folder, `my-repository`, and start the ``server`` script:
+
+.. code-block:: console
+
+    (my-site) $ cd ../my-site
+    (my-site) $ pipenv run ./scripts/server
```

Comment:
Since we are already in activated virtualenv (`my-site`), we should be able to run server script directly - ` ./scripts/server`